Functions/Admin/Backup-RsEncryptionKey.ps1

```diff
@@ -4,71 +4,134 @@
+	        Description
+	        -----------
+	        This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services
+	
+		.NOTES
+			Author:      ???
+			Editors:     Friedrich Weinmann
+			Created on:  ???
+			Last Change: 31.01.2017
+			Version:     1.1
+	
+			Release 1.1 (26.01.2017, Friedrich Weinmann)
+			- Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+			- Implemented ShouldProcess (-WhatIf, -Confirm)
+			- Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+			- Improved error message on failure.
+			- Standardized the parameters governing the Report Server connection for consistent user experience.
+	
+			Release 1.0 (???, ???)
+			- Initial Release
     #>
 
-    [cmdletbinding()]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param
     (
+        [Alias('SqlServerInstance')]
         [string]
-        $SqlServerInstance='MSSQLSERVER',
-
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
         [string]
-        $SqlServerVersion='13',
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential,
 
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
         $Password,
         
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
-        $KeyPath    
-    )
-
-    $rsWmiObject = New-RsConfigurationSettingObject -SqlServerInstance $SqlServerInstance -SqlServerVersion $SqlServerVersion
-
-    Write-Verbose "Retrieving encryption key..."
-    $encryptionKeyResult = $rsWmiObject.BackupEncryptionKey($Password)
-
-    if ($encryptionKeyResult.HRESULT -eq 0) 
-    {
-        Write-Verbose "Success!"
-    }
-    else 
-    {
-        Write-Error "Fail! `n Errors: $($encryptionKeyResult.ExtendedErrors)"
-        Exit 1
-    }
-
-    Write-Verbose "Writing key to file..."
-    [System.IO.File]::WriteAllBytes($KeyPath, $encryptionKeyResult.KeyFile)
-    Write-Verbose "Success!"
+        $KeyPath
+	)
+    
+    if ($ComputerName) { $tempComputerName = $ComputerName }
+    else { $tempComputerName = ([ReportingServicesTools.ConnectionHost]::ComputerName) }
+    if ($ReportServerInstance) { $tempInstanceName = $ReportServerInstance }
+    else { $tempInstanceName = ([ReportingServicesTools.ConnectionHost]::Instance) }
+        
+    if ($PSCmdlet.ShouldProcess("$tempComputerName \ $tempInstanceName", "Retrieve encryption key and create backup in $KeyPath"))
+	{
+		#region Connect to Report Server using WMI
+        try
+        {
+            $splat = @{ }
```

Comment:
I like the option of passing the $PSBoundParametersDictionary and return back the hashtable for the splat, in that way the function intention is clear about parsing the defaults and the current function is clear about getting the wmi object.
As name it could be something uninspired like 
$connectionInfo = Get-ConnectionInfoValuesOrDefault ($PSBoundParameters)
$rsWmiObject = New-RsConfigurationSettingObject $connectionInfo